examples/tv_minimization_reconstruction_pdhg.py

```diff
@@ -0,0 +1,190 @@
+# data.
+#
+# Let $y$ denote the k-space data of the image $x_{\mathrm{true}}$ sampled with an acquisition model $A$
+# (Fourier transform, coil sensitivity maps, ...), i.e the forward problem is given as
+#
+# $ y = Ax_{\mathrm{true}} + n, $
+#
+# where $n$ describes complex Gaussian noise. When using TV-minimization as regularization method, an approximation of
+# $x_{\mathrm{true}}$ is obtained by minimizing the following functional $\mathcal{F}$:
+#
+# $ \mathcal{F}(x) = \frac{1}{2}||Ax - y||_2^2 + \lambda \| \nabla x \|_1, \quad \quad \quad (1)$
+#
+# where $\nabla$ is the discretized gradient operator.
+#
+# The minimization of the functional $\mathcal{F}$ is a non-trivial task due to the presence of the operator
+# $\nabla$ in the non-differentiable $\ell_1$-norm. A suitable algorithm to solve the problem is the
+# PDHG-algorithm [Chambolle \& Pock, JMIV 2011].
+#
+# PDHG is a method for solving problems of the form
+#
+# $ \min_x f(K(x)) + g(x)  \quad \quad \quad (2)$
+#
+# where $f$ and $g$ denote proper, convex, lower-semicontinous functionals and $K$ denotes a linear operator.
+#
+# PDHG essentially consists of three steps, which read as
+#
+# $z_{k+1} = \mathrm{prox}_{\sigma f^{\ast}}(z_k + \sigma K \bar{x}_k)$
+#
+# $x_{k+1} = \mathrm{prox}_{\tau g}(x_k - \tau K^H z_{k+1})$
+#
+# $\bar{x}_{k+1} = x_{k+1} + \theta(x_{k+1} - x_k)$,
+#
+# where $\mathrm{prox}$ denotes the proximal operator and $f^{\ast}$ denotes the convex conjugate of the
+# functional $f$, $\theta\in [0,1]$ and step sizes $\sigma, \tau$ such that $\sigma \tau < 1/L^2$, where
+# $L=\|K\|_2$ is the operator norm of the operator $K$.
+#
+# The first step is to recast problem (1) into the general form of (2) and then to apply the steps above
+# in an iterative fashion.
+#
+# A possible and intuitive (but unfortunately not efficient) identification is the following
+#
+# $f(x) = \lambda \| x\|_1,$
+#
+# $g(x) = \frac{1}{2}\|Ax  - y\|_2^2,$
+#
+# $K(x) = \nabla x$,
+#
+# However, although $\mathrm{prox}_{\sigma f^\ast}$ has a simple form, some calculations show that
+# to be able to compute $\mathrm{prox}_{\tau g}$, one would need to solve a linear system at each
+# iteration.
+#
+# Thus, another (less intuitive, but way more efficient) identification is the following:
+#
+# $f(z) = f(p,q) = f_1(p) + f_2(q) =  \frac{1}{2}\|p  - y\|_2^2 + \lambda \| q \|_1,$
+#
+# $K(x) = [A, \nabla]^T,$
+#
+# $g(x) = 0,$
+#
+# for which one can show that both $\mathrm{prox}_{\sigma f^{\ast}}$ and $\mathrm{prox}_{\tau g}$ are
+# given by simple and easy-to-compute operations, see for example [Sidky et al, PMB 2012].
+#
+# In the following, we load some 2D radial MR data and use the just described identification to set up
+# the corresponding problem to be solved with PDHG.
+
+# %%
+import mrpro
+
+# %% [markdown]
+# ##### Read-in the raw data
+
+# %%
+# Use the trajectory that is stored in the ISMRMRD file
+trajectory = mrpro.data.traj_calculators.KTrajectoryIsmrmrd()
```

Comment:
I would not save this in a variable named trajectory. consider not saving it as a variable and only giving it as an input to from_file.

What you have there is not the trajectory, it only tells from_file how to get the trajectory.